azure-quantum/tests/quantum_test_base.py

```diff
@@ -0,0 +1,197 @@
+        regex_replacer.register_regex(r'/resourceGroups/[a-z0-9-]+/', "/resourceGroups/dummy-rg/")
+        regex_replacer.register_regex(r'/workspaces/[a-z0-9-]+/', "/workspaces/dummy-ws/")
+        regex_replacer.register_regex(r'sig=[0-9a-zA-Z%]+\&', "sig=sanitized&")
+        regex_replacer.register_regex(r'sv=[^&]+\&', "sv=sanitized&")
+        regex_replacer.register_regex(r'se=[^&]+\&', "se=sanitized&")
+
+    def setUp(self):
+        super(QuantumTestBase, self).setUp()
+        # mitigation for issue https://github.com/kevin1024/vcrpy/issues/533
+        self.cassette.allow_playback_repeats = True
+
+    @property
+    def client_id(self):
+        return self._client_id
+
+    @property
+    def client_secret(self):
+        return self._client_secret
+
+    @property
+    def tenant_id(self):
+        return self._tenant_id
+
+    @property
+    def resource_group(self):
+        return self._resource_group
+
+    @property
+    def subscription_id(self):
+        return self._subscription_id
+
+    @property
+    def workspace_name(self):
+        return self._workspace_name
+
+    def create_workspace(self) -> Workspace:
+        """Create workspace using credentials stored in config file
+
+        :return: Workspace
+        :rtype: Workspace
+        """
+
+        workspace = Workspace(
+        
+            subscription_id=self.subscription_id,
+            resource_group=self.resource_group,
+            name=self.workspace_name,
+        )
+
+        if self.is_live or self.in_recording:
+            workspace.credentials = ServicePrincipalCredentials(
+                tenant=self.tenant_id,
+                client_id=self.client_id,
+                secret=self.client_secret,
+                resource  = "https://quantum.microsoft.com"
+            )
+            workspace.login(False)
+        else:
+            workspace.credentials = BasicTokenAuthentication(token=self.dummy_auth_token)
+
+        return workspace
+
+class CustomRecordingProcessor(RecordingProcessor):
+
+    ALLOW_HEADERS = [
+        'content-length',
+        'content-type',
+        'accept',
+        'accept-encoding',
+        'accept-charset',
+        'accept-ranges',
+        'x-ms-range',
+        'transfer-encoding',
+        'x-ms-blob-content-md5',
+        'x-ms-blob-type',
+        'x-ms-creation-time',
+        'x-ms-lease-state',
+        'x-ms-lease-status',
+        'x-ms-server-encrypted',
+        'x-ms-version'
+        ]
+    
+    def __init__(self):
+        self._regexes = []
+
+    def register_regex(self, oldRegex, new):
+        self._regexes.append((re.compile(oldRegex), new))
+
+    def process_request(self, request):
+        headers = {}
+        for key in request.headers:
+            if key.lower() in self.ALLOW_HEADERS:
+                headers[key] = request.headers[key]
+        #request.headers = headers
```

Comment:
```suggestion
```

Reply:
I removed the comment for this line to re-enable filtering which headers we want in the recordings with PR #68.